### ISSUE TITLE
Complete Issue Publishing

### ISSUE THREAD

**Issue by justingonder** [openlibhums/janeway#1591]
**Is your feature request related to a problem? Please describe.**
As the editor of a legacy and/or print journal (or a journal that simply prefers to publish complete issues), I want the option to queue up my articles in a complete issue and then publish all of them at the same time.

**Describe the solution you'd like**
If the Journal Manager has indicated that the journal publishes only complete issues (perhaps via a Journal Settings toggle) then I would expect the current “Publish this Article” button on the Pre-Publication checklist page to go away, and instead in the space currently occupied by that button there would be some link to the relevant Issue Management page. I would click through to the Issue Management page where I would eventually see all of the articles queued up for the issue and once the entire issue is ready, I could push a button to publish it.

**Describe alternatives you've considered**
It’s theoretically possible to approximate this currently by simply not clicking the final “Publish this Article” button for all articles in a given issue, and then rapidly clicking that button for each individual article when you want to publish the complete issue.

**Additional context**
_Screenshots attached._
Some additional considerations:
- In addition to a ‘Publish Issue’ button on the Issue Management page, there might also need to be an ‘Update Published Issue’ button that appears instead of the publish issue button after an issue has initially been published. Editors would make use of this if they had made any adjustments to metadata or content in a published issue, and wanted to push those changes out.
- Journals that publish in this way often use a single year-only publication date for all articles in the issue. In this case, the “Set a Publication Date” item in the Pre Publication check-list would be unnecessary, and there would need to be a publication date field added to relevant Issue Management pages.

<img width="1514" alt="2020-05-27_14-40-32" src="https://user-images.githubusercontent.com/29953714/83070374-fd7e2b00-a030-11ea-9512-a33dbda1582d.png">
<img width="1509" alt="2020-05-27_14-40-07" src="https://user-images.githubusercontent.com/29953714/83070389-053dcf80-a031-11ea-934d-923e3841c923.png">

**Comment by justingonder** [comment]
hi @joemull - yes, this kind of functionality remains very relevant to the majority of our journals, though I wrote the above some time ago so am definitely open to other approaches.

**Comment by ajrbyers** [comment]
I have one additional thought on this: we'd need to remove the set publication date from prepub so it can only be altered through the issue interface.

**Comment by ajrbyers** [comment]
After a discussion with CDL on the best approach for this my suggestion is:

1. Add a new setting to toggle whether a journal publishes in issues
2. Update the Prepub page to remove the date/time field and change the publish button wording (note: this button should probably be hidden if there is no issue set)
3. Clicking what was the publish button will set the article pub date to the issue and move the article into the Published stage
4. Add a new signal to the Issue model so that when the pub date is changed, it updates the Issue's articles
5. Add a new signal that will set the article pub date to the issue pub date when adding an article to an issue, unless its in another issue/as a primary issue
6. Add a warning that changing the issue pub date will update all of the article pub dates

**Comment by pgoussy** [comment]
This sounds good for most use-cases, but (if I'm following the logic correctly) the pub date stuff would be problematic for journals that publish on a rolling schedule. Those journals typically have a single volume/issue for each calendar year, which lists a pub date of 01/01/20XX and then each individual article gets its own "actual" pub date as they go live throughout the year. See, for instance, [PTPBio](https://journals.publishing.umich.edu/ptpbio/issue/73/info/) - for these types of journals, we don't want the article pub date to automatically be set to the issue pub date.

**Comment by ajrbyers** [comment]
> This sounds good for most use-cases, but (if I'm following the logic correctly) the pub date stuff would be problematic for journals that publish on a rolling schedule. Those journals typically have a single volume/issue for each calendar year, which lists a pub date of 01/01/20XX and then each individual article gets its own "actual" pub date as they go live throughout the year. See, for instance, [PTPBio](https://journals.publishing.umich.edu/ptpbio/issue/73/info/) - for these types of journals, we don't want the article pub date to automatically be set to the issue pub date.

Hey @pgoussy disabling the setting in step 1 would allow continuous publication journals to go along as usual with articles having their own ongoing pub dates.
